src/basicfuns.jl

```diff
@@ -259,12 +254,38 @@ $(SIGNATURES)
 Return the [`softmax transformation`](https://en.wikipedia.org/wiki/Softmax_function)
```

Comment:
Also mention `dims` here and below? While you're at it, it would make sense to use the same wording as above, which is more explicit.

Reply:
I updated the docstrings and merged the ones for the inplace method. Also I noticed that `$(SIGNATURES)` does not include the default value of the keyword arguments, and hence I wrote the signature manually.